### DISCUSSION TITLE
Android - Error retrieving information from server df-dferh-01

### DISCUSSION THREAD

**Discussion by MC-HaoPhan** [hyochan/react-native-iap#1990] [Q&A] [answered]
I have a problem when I upgrade iap version from 6.0.6 to 9.0.4. In the dev version after calling requestSubscription I see true return. But when I build to the test version, I get the error "Error retrieving information from server df-dferh-01".
- react-native-iap: 9.0.4
- react-native: 0.65.2
- Platforms (iOS, Android, emulator, simulator, device): Samsung Galaxy S7

**Comment by MC-HaoPhan** [comment]
Then more errors show up: 
**Purchase error 
The service is unavailable. This may internet connection, or the Play Store may be down.**

**Comment by andresesfm** [comment, marked as answer]
his is not related to the library https://support.google.com/googleplay/thread/81842487/error-retrieving-information-from-server-df-dferh-01?hl=en

**Reply by MC-HaoPhan** [reply]
Thank you so much

**Comment by normenster** [comment]
@andresesfm I've implemented the 12.3.2 Version and it worked well with my google account. We have the app on an test track in play console, assigned several users and added them to license test. All users joined the test program with the link. Now all test users see the subscriptions, but when clicking requestSubscription they get the df-dferh-01 Error. I`ve have created a separate user account and can reproduce that behaviour on my device with that new account. I did everything including factory reset, reset caches, adding billing information, pin code, 2fa, tried different react-native-iap versions - It doesnt help. When i switch back to my live google account, it starts working again. Is there some hint or trick what the user account needs to have to work with the google billing service in the sandbox szenario?

**Comment by normenster** [comment]
@andresesfm for me it was a wrong call to requestSubscription. We have Subs with 7 days trial and the subscription. 
`if (Platform.OS === 'android') {
                      subscription.subscriptionOfferDetails.map((sub) => {
                        offers.push({
                          sku: subscription.productId,
                          offerToken: sub.offerToken,
                        });
                      });
                    }
`

i thought i have to send all offers:

` const subme = await requestSubscription({
                      sku: subscription.productId,
                      subscriptionOffers: offers
                    });`


but that creates the error for some users ( i don`t know why it worked with my account).
Only the "real" sub needs to be sent - this worked then: 

` const subme = await requestSubscription({
                      sku: subscription.productId,
                      subscriptionOffers: [offers[0]],
                    });`
